tls/s2n_config.c

```diff
@@ -1082,6 +1089,13 @@ int s2n_config_set_renegotiate_request_cb(struct s2n_config *config, s2n_renegot
     return S2N_SUCCESS;
 }
 
+S2N_RESULT s2n_config_set_setsockopt(struct s2n_config *config, s2n_setsockopt_cb cb)
+{
+    RESULT_ENSURE_REF(config);
```

Comment:
If this is only used in testing, probably worth adding a check to confirm that

Reply:
unfortunately we do need to use it to configure the default setsockopt.. but i could set it directly rather than call this fn and then this can have the check :+1: